### PR TITLE
DDF-4812 Fix user sessions expiring prematurely

### PR DESCRIPTION
**Pull request by lamhuy**
**What does this PR do?**
Updates the "expiry" for the session across all contexts, so that the session is refreshed in all contexts.

**Who is reviewing it?**
@kcover @ahoffer @tyler30clemens @austinsteffes

Ask 2 committers to review/merge the PR and tag them here.
@clockard
@stustison
@troymohl

**How should this be tested?**

- Install DDF
- Go to Intrigue and leave it idle for 11 minutes
- Make sure no red dialog boxes pop up saying "logged out" from the 9-11 minute mark
- Any background context you want to provide?
- What are the relevant tickets?

Fixes: #4812

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by lamhuy** [comment]
build now

**Comment by brjeter** [comment]
@lamhuy: To address @pklinef's comment, I would recommend adding a test to the https://github.com/codice/ddf/blob/master/platform/platform-paxweb-jettyconfig/src/test/java/org/codice/ddf/security/session/JettySessionManagementTest.java that lowers the Jetty Housekeeper's scavenging config property to a few seconds and make sure ongoing requests keep the session alive across all contexts.

**Comment by lamhuy** [comment]
build now

**Comment by lamhuy** [comment]
build now

**Comment by ahoffer** [comment]
Build failed with a check-style error. The variable `MAX_INACTIVE_INTERVAL_SECOND` should, I think be in camel case because it is not `final`

09:47:42 [ERROR] /jenkins/workspace/DDF-Jobs/pr/Linux/platform/platform-paxweb-jettyconfig/src/test/java/org/codice/ddf/security/session/JettySessionManagementTest.java:47:22: Name 'MAX_INACTIVE_INTERVAL_SECOND' must match pattern '^[a-z][a-zA-Z0-9]*$'. [StaticVariableName]

**Comment by lamhuy** [comment]
build now

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingSessionDataStore.java [hunk 1]
```diff
@@ -46,19 +46,20 @@
...
-   * @param sessionAttributes the session's attributes
+   * @param updatedSessionData the session's attributes
    */
-  public void updateSessionAttributes(String id, Map<String, Object> sessionAttributes) {
+  public void updateSessionAttributes(String id, SessionData updatedSessionData) {
     synchronized (sessionDataMap) {
       SessionData sessionData = sessionDataMap.get(id);
 
-      if (sessionData != null && !sessionData.getAllAttributes().equals(sessionAttributes)) {
+      if (sessionData != null && !sessionData.getAllAttributes().equals(sessionData)) {
```
Comment (brjeter) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingSessionDataStore.java]: ✏️  I just noticed that this should be `updatedSessionData.getAllAttributes()`, not `sessionData`.
It's not a big deal, just a minor performance tweak, so I think we can address this in a follow-on PR.

**Comment by lamhuy** [comment]
build now

**Comment by brjeter** [comment]
Hero successful ✅

**Pull request merged**